components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,586 @@
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+        );
+        action = result;
+    }
+
+    @Test(priority = 8)
+    public void testUpdateActionWithoutNameAndDescription() throws ActionMgtException {
+
+        Action updatingAction = buildMockUpdatingActionWithBasicAuth(
+                null,
+                null,
+                "https://sample.com",
+                "updatingadmin",
+                "updatingadmin");
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+                           );
+    }
+
+    @Test(priority = 9)
+    public void testUpdateActionWithoutEndpointConfiguration() throws ActionMgtException {
+
+        Action updatingAction = buildMockUpdatingActionWithoutEndpointConfiguration();
+        Action result = daoImpl.updateAction(PRE_ISSUE_ACCESS_TOKEN, action.getId(), updatingAction, action, TENANT_ID);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                action.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+                           );
+    }
+
+    @Test(priority = 10)
+    public void testUpdateActionEndpointAuthProperties() {
+
+        try {
+            Action result = daoImpl.updateActionEndpointAuthProperties(action.getId(),
+                    buildMockBasicAuthType(true),
+                    TENANT_ID);
+            Assert.assertEquals(AuthType.AuthenticationType.BASIC, result.getEndpoint().getAuthentication().getType());
+            Assert.assertEquals(2, result.getEndpoint().getAuthentication().getProperties().size());
```

Comment:
Assert for values. May be it's an alias if so assert for the alias

Reply:
I am not totally convinced by the expected change. 

Is it something like rather than asserting AuthenticationType , should we assert for values like String, int ?

The suggested change --->

Assert.assertEquals(AuthType.AuthenticationType.BASIC.getType(), result.getEndpoint().getAuthentication().getType().getType())